mail_template_restrict_by_default/hooks.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2024 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+from odoo import SUPERUSER_ID, api
+
+
+def post_init_hook(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    env["ir.config_parameter"].set_param("mail.restrict.template.rendering", True)
+    env.ref("mail.group_mail_template_editor").write(
```

Comment:
this shouldn't be needed as long as the admin is in `group_system`

Reply:
it's meant to remove other users. Didn't debug it yet, but when checking the flag on the [current runbot](https://59019060-17-0-all.runbot211.odoo.com/web?debug=1&reload=true#action=95&model=res.config.settings&view_type=form&cids=1&menu_id=70), it breaks the implied_group relation, but keeps the demo user in the [group](https://59019060-17-0-all.runbot211.odoo.com/web?debug=1#id=14&cids=1&menu_id=59&action=68&model=res.groups&view_type=form)

Reply:
ah, that's because email marketing/user inherits this too. not sure what to do about that, let's see what others think

Reply:
Ugh it's in https://github.com/odoo/odoo/blob/16.0/addons/mail/models/ir_config_parameter.py#L25-L35

Reply:
I would expect a module by this name to apply the restriction globally, so removing the group as an implied group everywhere apart from the system group.

Reply:
Override res.group's write method and intercept these writes when you encounter the `install_module` context key?

Reply:
Note that the "default user has too many rights" was handled recently in odoo/odoo#150391 (odoo/odoo#149424 for master) if you are not in a "trust everyone by default" mode.

Reply:
Yes I meant to thank you for that :smile: but that does not solve the implied groups of base.groups_user I think.

Reply:
@StefanRijnhart about suppressing writes that set the implied group: I think better not, the first thing I will do after adding it to my ${customer}_base module, is adding the template editor group as implied group to the customers 'Trusted powerusers' group who are the kind of people who may edit mail templates.

It is a lot of fluff for the little the modules actually do, but I think we'll be better off with only removing things explicitly.